src/main/java/duke/command/FindCommand.java

```diff
@@ -0,0 +1,39 @@
+package duke.command;
+
+import duke.Storage;
+import duke.TaskList;
+import duke.ui.Ui;
+
+/**
+ * Find tasks from the TaskList with the specified keyword.
+ */
+public class FindCommand extends Command {
+    private String keyword;
+
+    /**
+     * Constructs a FindCommand with the specified keyword.
+     *
+     * @param keyword The keyword to filter.
+     */
+    public FindCommand(String keyword) {
+        this.keyword = keyword;
+    }
+
+    /**
+     * Finds tasks with the specified keyword and returns a string representation of the tasks.
+     *
+     * @param tasks The TaskList containing the tasks.
+     * @param ui The Ui that interacts with the user.
+     * @param storage The Storage to load and save tasks into the data file.
+     * @return A string with the message to be printed.
+     */
+    @Override
+    public String execute(TaskList tasks, Ui ui, Storage storage) {
```

Comment:
One more thing take note for future implementation: It is not generally recommended to pass UI and Storage object to all the classes and update it in multiple places. It reduces code readability and maintainability. You will learn more related principles in upcoming lectures. But keep this in your mind so that you can related it when you learn